### DISCUSSION TITLE
Duplicate GlobalKey detected in widget tree.

### DISCUSSION THREAD

**Discussion by markarik** [rrousselGit/riverpod#1269] [Q&A]
When using HooksConsumerWidget am getting this error here
════════ Exception caught by widgets library ═══════════════════════════════════
The following assertion was thrown while finalizing the widget tree:
Duplicate GlobalKey detected in widget tree.

The following GlobalKey was specified multiple times in the widget tree. This will lead to parts of the widget tree being truncated unexpectedly, because the second time a key is seen, the previous instance is moved to the new location. The key was:
- [LabeledGlobalKey<FormState>#0fc05]
This was determined by noticing that after the widget with the above global key was moved out of its previous parent, that previous parent never updated during this frame, meaning that it either did not update at all or updated before the widget was moved, in either case implying that it still thinks that it should have a child with that global key.
The specific parent that did not update after having one or more children forcibly removed due to GlobalKey reparenting is:
- Column(direction: vertical, mainAxisAlignment: center, crossAxisAlignment: center, renderObject: RenderFlex#e7b06 relayoutBoundary=up14 NEEDS-PAINT)
A GlobalKey can only be specified on one widget at a time in the widget tree.
When the exception was thrown, this was the stack
#0      BuildOwner.finalizeTree.<anonymous closure>

```import "package:flutter/material.dart";
import "package:flutter_form_builder/flutter_form_builder.dart";
import "package:infomobyagent/utils/constants/colors.dart";
import "package:infomobyagent/utils/widgets/shared_widgets.dart";
final _formKey = GlobalKey<FormState>();
class LoginRegisterPage extends HookConsumerWidget {
  const LoginRegisterPage({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final height = MediaQuery.of(context).size.height;
    return Scaffold(
        body: Center(
      child: Container(
        margin: EdgeInsets.only(top: height * .1),
        padding: const EdgeInsets.symmetric(horizontal: 20),
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              titleTitle(
                context,
                mainColor,
              ),
              const SizedBox(height: 20),
              FormBuilder(
                key: _formKey,
                child: Column(
                  children: const [],
                ),
              ),
            ],
          ),
        ),
      ),
    ));
  }
}
 ```

While using  statefulwidget the form is working  without giving the error 
```import "package:flutter/material.dart";
import "package:flutter_form_builder/flutter_form_builder.dart";
import "package:infomobyagent/utils/constants/colors.dart";
import "package:infomobyagent/utils/widgets/shared_widgets.dart";
class LoginRegisterPage extends StatefulWidget {
  const LoginRegisterPage({Key? key}) : super(key: key);
  @override
  LoginRegisterPageState createState() {
    return LoginRegisterPageState();
  }
}
class LoginRegisterPageState extends State<LoginRegisterPage> {
  final _formKey = GlobalKey<FormState>();
  @override
  Widget build(BuildContext context) {
    final height = MediaQuery.of(context).size.height;
    return Scaffold(
        body: Center(
      child: Container(
        margin: EdgeInsets.only(top: height * .1),
        padding: const EdgeInsets.symmetric(horizontal: 20),
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              titleTitle(
                context,
                mainColor,
              ),
              const SizedBox(height: 20),
              FormBuilder(
                key: _formKey,
                child: Column(
                  children: const [],
                ),
              ),
            ],
          ),
        ),
      ),
    ));
  }
}```

**Comment by rrousselGit** [comment]
Could you please format the code and use code blocks?

Riverpod is unrelated to keys anyway, so the problem is most definitely unrelated to Riverpod

**Reply by markarik** [reply]
Sorry for the unformated code, have alreaady formated it,,my issue comes in when am using HookConsumerWidget

**Reply by rrousselGit** [reply]
Are you maybe using LoginRegisterPage twice, or is the route present multiple times in your navigation history?
The problem is that in the consumer variant, your GlobalKey is global, so shared between all instances of LoginResiterPage

You'd have the same problem with StatefulWidget if you made the key global for it too

**Comment by iamaryanshandilya** [comment]
> Are you maybe using LoginRegisterPage twice, or is the route present multiple times in your navigation history? The problem is that in the consumer variant, your GlobalKey is global, so shared between all instances of LoginResiterPage
> 
> You'd have the same problem with StatefulWidget if you made the key global for it too

can you please tell me how do I resolve this, I am using it twice, and I really dont understand what should I do? sorry I know this isnt related to riverpod, but this is the closed to the error, I am getting, and your reply made a lot of sense although I dont understand how do I resolve it
